nuitka/tools/testing/SearchModes.py

```diff
@@ -82,10 +82,24 @@ def _match(cls, dirname, filename, candidate):
 
         return candidate.rstrip("/") in candidates or candidate2 in candidates
 
+    def exit(self, message):
+        # Virtual method, pylint: disable=no-self-use
+        sys.exit(message)
+
+    def isSearchModeAll(self):
+        # Virtual method, pylint: disable=no-self-use
+        return False
+
     def isCoverage(self):
         # Virtual method, pylint: disable=no-self-use
         return False
 
+    def onErrorDetected(self, message):
+        if self.isSearchModeAll():
```

Comment:
Please use overloading of methods. This should simply be the `sys.exit` I figure, while the method in the all search mode would do what it does. You do not need `isSearchModeAll` in this case, which was the point of using a method.